content/5_Plat/HLD.mdx

```diff
@@ -252,7 +252,137 @@ set problems where $\Theta(Q\sqrt N\log N)$ is intended ...
 	</Resource>
 </Resources>
 
-<IncompleteSection />
+
+<!-- <IncompleteSection /> -->
+
+The heavy-light decomposition algorithm is versatile and can be combined with other data structures like segment trees, binary indexed trees, and range minimum query data structures to solve a wide range of complex tree-related problems.
+
+## Template
+
+<LanguageSection>
+<CPPSection>
+```cpp
+
+// BeginCodeSnip{Segment Tree}
+struct Segtree {
```

Comment:
im p sure we have a RMQ template [here](https://usaco.guide/gold/PURS?lang=cpp), altho now that I think abt it this template doesn't have O(N) init lmao